### ISSUE TITLE
Godot 4 beta 8 mono crash on opening project.

### ISSUE THREAD

**Issue by lohmaster** [godotengine/godot#70129]
### Godot version

4.0 beta 8

### System information

Windows 11, any rendering backend, gtx 1060

### Issue description

Godot output:
ERROR: hostfxr_initialize_for_runtime_config failed with code: -2147450733
   at: (modules/mono/mono_gd/gd_mono.cpp:201)
ERROR: Parameter "load_assembly_and_get_function_pointer" is null.
   at: initialize_hostfxr_and_godot_plugins (modules/mono/mono_gd/gd_mono.cpp:274)
ERROR: Parameter "godot_plugins_initialize" is null.
   at: initialize (modules/mono/mono_gd/gd_mono.cpp:403)

dotnet -- info output:
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Środowisko uruchomieniowe:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  6.0.403 [C:\Program Files\dotnet\sdk]
  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

### Steps to reproduce

Open any project.

### Minimal reproduction project

N/A

**Comment by adamscott** [comment]
Citing [CONTRIBUTING.md](https://github.com/godotengine/godot/blob/e032152494646da7933fbef13d2914c67b09ce9a/CONTRIBUTING.md#reporting-bugs)

> On the contributors' end, recreating a test project from scratch takes valuable time that can be saved by uploading a _minimal_ project.

> To speed up our work, **please upload a minimal project** that isolates and reproduces the issue. This is always the **best way for us to fix it**. We recommend attaching a ZIP file with the minimal project directly to the bug report, by drag and dropping the file in the GitHub edition field. This ensures the file can remain available for a long period of time. Only use third-party file hosts if your ZIP file isn't accepted by GitHub because it's too large.

> We recommend always attaching a minimal reproduction project, even if the issue may seem simple to reproduce manually.

Please join a MRP.

**Comment by adamscott** [comment]
And please use [code quotation marks](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) when citing code. Please edit your description.

**Comment by akien-mga** [comment]
@adamscott If it crashes when opening a brand new project, a MRP isn't really needed.

This sounds related to #68652, it's probably caused by the installed .NET Core 7. CC @godotengine/dotnet

**Comment by raulsntos** [comment]
> ERROR: hostfxr_initialize_for_runtime_config failed with code: -2147450733

Acording to the error code (see https://github.com/dotnet/runtime/blob/main/docs/design/features/host-error-codes.md), the error received was `InvalidConfigFile` with description:

> The `.runtimeconfig.json` file is invalid. The reasons for this failure can be among these:
>
> - Failure to read from the file
> - Invalid JSON
> - Invalid value for a property (for example number for property which requires a string)
> - Missing required property
> - Other inconsistencies (for example `rollForward` and `applyPatches` are not allowed to be specified in the same config file)
> - Any of the above failures reading the `.runtimecofig.dev.json` file
> - Self-contained `.runtimeconfig.json` used in `hostfxr_initialize_for_runtime_config`. Note that missing `.runtimconfig.json` is not an error (means self-contained app). This error code is also used when there is a problem reading the CLSID map file in `comhost`.

**Comment by Markyparky56** [comment]
Compiling latest (5674fa04aeb44a0d9cf48073fe2be25d1b804acf) I am also seeing this CTD on launching a blank project. Compiled the editor in dev mode and caught it in the debugger (VS2022). Trickier than expected because the initially attached/launched application is not the one that crashes.

Callstack:
```0000000000000000()
godot.windows.editor.dev.x86_64.mono.exe!CSharpLanguage::_editor_init_callback() Line 1128
	at C:\godot\modules\mono\csharp_script.cpp(1128)
godot.windows.editor.dev.x86_64.mono.exe!EditorNode::EditorNode() Line 7421
	at C:\godot\editor\editor_node.cpp(7421)
godot.windows.editor.dev.x86_64.mono.exe!Main::start() Line 2810
	at C:\godot\main\main.cpp(2810)
godot.windows.editor.dev.x86_64.mono.exe!widechar_main(int argc, wchar_t * * argv) Line 178
	at C:\godot\platform\windows\godot_windows.cpp(178)
godot.windows.editor.dev.x86_64.mono.exe!_main() Line 203
	at C:\godot\platform\windows\godot_windows.cpp(203)
godot.windows.editor.dev.x86_64.mono.exe!main(int argc, char * * argv) Line 217
	at C:\godot\platform\windows\godot_windows.cpp(217)
godot.windows.editor.dev.x86_64.mono.exe!WinMain(HINSTANCE__ * hInstance, HINSTANCE__ * hPrevInstance, char * lpCmdLine, int nCmdShow) Line 231
	at C:\godot\platform\windows\godot_windows.cpp(231)
[Inline Frame] godot.windows.editor.dev.x86_64.mono.exe!invoke_main() Line 102
	at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl(102)
godot.windows.editor.dev.x86_64.mono.exe!__scrt_common_main_seh() Line 288
	at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl(288)
kernel32.dll!00007ffa8fb67614()
ntdll.dll!00007ffa902426a1()
```

mini dump attached, full heap dump available but is probably too large to upload here even if I zip it. 
[godot.windows.editor.dev.x86_64.mono.mini.dmp.zip](https://github.com/godotengine/godot/files/10316993/godot.windows.editor.dev.x86_64.mono.mini.dmp.zip)

Machine is in a similar state as the OP. .NET 6.0 SDK and 7.0 side-by-side

``` C:\godot>dotnet --info
.NET SDK:
 Version:   7.0.101
 Commit:    bb24aafa11

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.101\

Host:
  Version:      7.0.1
  Architecture: x64
  Commit:       97203d38ba

.NET SDKs installed:
  6.0.404 [C:\Program Files\dotnet\sdk]
  7.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found
```

**Comment by lohmaster** [comment]
It seems after last update of Visual Studio this issue went away. I am not sure whether it had anything to do with it. It is however the only software change that happened on my system since I posted original issue.

**Comment by akien-mga** [comment]
Crashes with .NET 7 specifically should be fixed by #71825.
